packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts

```diff
@@ -50,4 +43,68 @@ export class DatasyncMongoDBDataProvider<Type = any, GraphbackContext = any> ext
 
     throw new NoDataError(`Could not find document to update in ${this.collectionName}`);
   }
+
+  public async delete(data: any): Promise<Type> {
+
+    const { idField } = getDatabaseArguments(this.tableMap, data);
+
+    if (!idField.value) {
+      throw new NoDataError(`Cannot delete ${this.collectionName} - missing ID field`)
+    }
+
+    data = {};
+
+    this.fieldTransformMap[TransformType.UPDATE]
+      .forEach((f: FieldTransform) => {
+        data[f.fieldName] = f.transform();
+      });
+
+    data._deleted = true;
+
+    const result = await this.db.collection(this.collectionName).updateOne({ _id: new ObjectId(idField.value), _deleted: false }, { $set: data });
+    if (result.result.nModified === 1) {
+      const queryResult = await this.db.collection(this.collectionName).find({ _id: new ObjectId(idField.value) }).toArray();
```

Comment:
the `findOne` method will miss any soft deleted items, since we are fetching the item that we just soft deleted, I don't think we can use findOne here